Bun.secrets.get not working on linux fedora 42 : panic: Segmentation fault at address 0x26978C0

### How can we reproduce the crash?

Hello, I m using Bun secrets on fedora 42

this code seems to be crashing the code : 
```js
authToken = await secrets.get({
                    service: "geekmusic-ytparser",
                    name: "authentification-token",
                })
```

secrets is Bun.secrets

there's an authentification set up. I guess it's a bug

It works on windows 11 with the same version

### Relevant log output

```shell
============================================================
Bun v1.2.21 (7c45ed97) Linux x64
Linux Kernel v6.15.7 | glibc v2.41
CPU: sse42 popcnt avx avx2
Args: "bun" "run" "main.js"
Features: Bun.stdin(2) Bun.stdout(2) WebSocket fetch jsc spawn tsconfig 
Builtins: "bun:main" "node:events" "node:fs" "node:os" "node:path" "node:readline" 
Elapsed: 33790ms | User: 282ms | Sys: 205ms
RSS: 1.10GB | Peak: 78.42MB | Commit: 1.10GB | Faults: 0 | Machine: 15.45GB

panic: Segmentation fault at address 0x26978C0
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.2.21/lr17c45ed9Q8gkwggCklyuvE+iuGogtH+6uJw2wJy4xJm50mlEiu1qpDujgmiDm9ocA2Ags8ltC

Illegal instruction (core dumped)
```

### Stack Trace (bun.report)

Bun v1.2.21 ([`7c45ed9`](<https://github.com/oven-sh/bun/tree/7c45ed97def1264717a1b55ab0f789f2ea58f986>)) on linux x86_64 [RunCommand]

Segmentation fault at address 0x026978C0

- [`SecretsLinux.cpp:297`](<https://github.com/oven-sh/bun/blob/7c45ed97def1264717a1b55ab0f789f2ea58f986/src/bun.js/bindings/SecretsLinux.cpp#L297>): `Bun__SecretsJobOptions__runTask`
- [`JSSecrets.zig:37`](<https://github.com/oven-sh/bun/blob/7c45ed97def1264717a1b55ab0f789f2ea58f986/src/bun.js/bindings/JSSecrets.zig#L37>): `bun.js.bindings.JSSecrets.SecretsJob.runTask`
- [`ThreadPool.zig:580`](<https://github.com/oven-sh/bun/blob/7c45ed97def1264717a1b55ab0f789f2ea58f986/src/threading/ThreadPool.zig#L580>): `Thread.PosixThreadImpl.spawn.Instance.entryFn`

Features: tsconfig, Bun.stdin, Bun.stdout, WebSocket, fetch, jsc, spawn

<!-- from bun.report: 3MvGBt2hcDRGuRRjZ3FV -->

<!-- sentry_id: c5dbabe4e5c62acc9454f0e4d068b451 -->

--- a/src/bun.js/bindings/SecretsLinux.cpp
+++ b/src/bun.js/bindings/SecretsLinux.cpp
@@ -240,7 +240,7 @@ static LibsecretFramework* libsecretFramework()
 // Define our simple schema for Bun secrets
 static const SecretSchema* get_bun_schema()
 {
-    static const SecretSchema schema = {
+    static SecretSchema schema = {
         "com.oven-sh.bun.Secret",
         SECRET_SCHEMA_NONE,
         { { "service", SECRET_SCHEMA_ATTRIBUTE_STRING },

--- a/src/bun.js/bindings/SecretsLinux.cpp
+++ b/src/bun.js/bindings/SecretsLinux.cpp
@@ -25,6 +25,7 @@ typedef int gboolean;
 typedef char gchar;
 typedef void* gpointer;
 typedef unsigned int guint;
+typedef int gint;
 
 // GLib constants
 #define G_FALSE 0
@@ -50,6 +51,14 @@ struct _SecretSchema {
     const gchar* name;
     SecretSchemaFlags flags;
     SecretSchemaAttribute attributes[32];
+    gint reserved;
+	gpointer reserved1;
+	gpointer reserved2;
+	gpointer reserved3;
+	gpointer reserved4;
+	gpointer reserved5;
+	gpointer reserved6;
+	gpointer reserved7;
 };
 
 struct _GError {
@@ -240,7 +249,7 @@ static LibsecretFramework* libsecretFramework()
 // Define our simple schema for Bun secrets
 static const SecretSchema* get_bun_schema()
 {
-    static SecretSchema schema = {
+    static const SecretSchema schema = {
         "com.oven-sh.bun.Secret",
         SECRET_SCHEMA_NONE,
         { { "service", SECRET_SCHEMA_ATTRIBUTE_STRING },

--- a/src/bun.js/bindings/SecretsLinux.cpp
+++ b/src/bun.js/bindings/SecretsLinux.cpp
@@ -51,6 +51,8 @@ struct _SecretSchema {
     const gchar* name;
     SecretSchemaFlags flags;
     SecretSchemaAttribute attributes[32];
+
+    /* <private> */
     gint reserved;
     gpointer reserved1;
     gpointer reserved2;
@@ -135,9 +137,6 @@ public:
         void* cancellable,
         GError** error);
 
-    // Collection name constant
-    const gchar* SECRET_COLLECTION_DEFAULT;
-
     LibsecretFramework()
         : secret_handle(nullptr)
         , glib_handle(nullptr)
@@ -221,13 +220,6 @@ private:
         secret_value_unref = (void (*)(gpointer))dlsym(secret_handle, "secret_value_unref");
         secret_item_get_attributes = (GHashTable * (*)(SecretItem*)) dlsym(secret_handle, "secret_item_get_attributes");
         secret_item_load_secret_sync = (gboolean(*)(SecretItem*, void*, GError**))dlsym(secret_handle, "secret_item_load_secret_sync");
-
-        // Load constants
-        void* ptr = dlsym(secret_handle, "SECRET_COLLECTION_DEFAULT");
-        if (ptr)
-            SECRET_COLLECTION_DEFAULT = *(const gchar**)ptr;
-        else
-            SECRET_COLLECTION_DEFAULT = "default";
 
         return g_error_free && g_free && g_hash_table_new && g_hash_table_destroy && g_hash_table_lookup && g_hash_table_insert && g_list_free && secret_password_store_sync && secret_password_lookup_sync && secret_password_clear_sync && secret_password_free;
     }

--- a/src/bun.js/bindings/SecretsLinux.cpp
+++ b/src/bun.js/bindings/SecretsLinux.cpp
@@ -39,7 +39,8 @@ typedef enum {
 
 typedef enum {
     SECRET_SCHEMA_ATTRIBUTE_STRING = 0,
-    SECRET_SCHEMA_ATTRIBUTE_INTEGER = 1
+    SECRET_SCHEMA_ATTRIBUTE_INTEGER = 1,
+    SECRET_SCHEMA_ATTRIBUTE_BOOLEAN = 2
 } SecretSchemaAttributeType;
 
 typedef struct {

--- a/src/bun.js/bindings/SecretsLinux.cpp
+++ b/src/bun.js/bindings/SecretsLinux.cpp
@@ -52,13 +52,13 @@ struct _SecretSchema {
     SecretSchemaFlags flags;
     SecretSchemaAttribute attributes[32];
     gint reserved;
-	gpointer reserved1;
-	gpointer reserved2;
-	gpointer reserved3;
-	gpointer reserved4;
-	gpointer reserved5;
-	gpointer reserved6;
-	gpointer reserved7;
+    gpointer reserved1;
+    gpointer reserved2;
+    gpointer reserved3;
+    gpointer reserved4;
+    gpointer reserved5;
+    gpointer reserved6;
+    gpointer reserved7;
 };
 
 struct _GError {

